add collaboratorpermission()

ianshmean opened the pull request:
Returns either `admin`, `write`, `read`, or `none` in accordance with https://developer.github.com/v3/repos/collaborators/#review-a-users-permission-level

ViralBShah commented:
Should we merge this?

ianshmean commented:
From my perspective I don't see a reason not to

ViralBShah commented:
Not sure who has the commit access who can get it merged. But let's give it another day or two now that it is bumped.

DilumAluthge commented:
@christopher-dG

christopher-dG commented:
Reviewing from my phone but have we tested this? It seems like `gh_get_json` doesn't return pagination data and the `result` would be some kind of Dict rather than an array (we're taking `result[2]` for reasons that are unclear to me). 

Please correct me if I'm way off base here.

DilumAluthge commented:
@ianshmean rebase on master?

ianshmean commented:
Rebased and renamed to conform to other function names.

Confirmed to work in 1.5.3
```
julia> collaborator_permission("JuliaIO/VideoIO.jl", "ianshmean", auth = myauth)
"admin"

julia> collaborator_permission("JuliaLang/julia", "ianshmean", auth = myauth)
ERROR: Error found in GitHub reponse:
	Status Code: 403
	Message: Must have push access to view collaborator permission.
	Docs URL: https://docs.github.com/rest/reference/repos#get-repository-permissions-for-a-user
```

I'm only doing this because I started it. It evidently wasn't needed for https://github.com/JuliaRegistries/Registrator.jl/pull/120

I'll let others judge if it's needed/correct

DilumAluthge commented:
LGTM

christopher-dG commented:
Huh, apparently I have no idea how this library works.
